x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/RequestUtils.java

```diff
@@ -10,13 +10,32 @@
 import org.apache.http.Header;
 import org.apache.http.HttpHeaders;
 import org.apache.http.message.BasicHeader;
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.common.CheckedSupplier;
+import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.rest.RestStatus;
+
+import java.net.URI;
+import java.net.URISyntaxException;
 
 public class RequestUtils {
 
     public static Header createAuthBearerHeader(SecureString apiKey) {
         return new BasicHeader(HttpHeaders.AUTHORIZATION, "Bearer " + apiKey.toString());
     }
 
+    public static URI buildUri(URI accountUri, String service, CheckedSupplier<URI, URISyntaxException> uriBuilder) {
+        try {
+            return accountUri == null ? uriBuilder.get() : accountUri;
+        } catch (URISyntaxException e) {
+            throw new ElasticsearchStatusException(
+                Strings.format("Failed to construct %s URL", service),
+                RestStatus.INTERNAL_SERVER_ERROR,
```

Comment:
> Is there a reason we don't log the URL here, but only the service name?

Yeah we probably should just to be safe.

> Are we expecting that the service URL should have already been validated by this point, and ever getting to this point in the code is the exception?

Yeah, if we actually get an exception I would expect it to be a programmatic error. It would be generated by a method like this one:

```
    static URI buildDefaultUri() throws URISyntaxException {
        return new URIBuilder().setScheme("https")
            .setHost(CohereUtils.HOST)
            .setPathSegments(CohereUtils.VERSION_1, CohereUtils.EMBEDDINGS_PATH)
            .build();
    }
```

Which essentially just uses static strings and calls the apache library. There's no user input. So unless one of those strings is like `^^` which we hardcode in, we really shouldn't get an exception.

Reply:
Actually, since we're passing in the `URISyntaxException` into the `ElasticsearchStatusException`. I believe it'll generate a `caused_by` section that gets returned which would include the invalid URL.

Reply:
We talked about this more offline. I'll switch this one to a bad request and then create a follow up PR to remove places where we log the URL and try to think of some other information we could log to help us with triaging errors. Maybe we could hash the url or something and log that instead.